src/backend/replication/logical/message.c

```diff
@@ -93,3 +96,62 @@ logicalmsg_redo(XLogReaderState *record)
 
 	/* This is only interesting for logical decoding, see decode.c. */
 }
+
+/*
+ * NEON: persist file in WAL to save it in persistent storage.
+ * If fd < 0, then remote entry from page server.
+ */
+void
+wallog_file_descriptor(char const* path, int fd, uint64_t limit)
+{
+	char	prefix[MAXPGPATH];
+
+	/* Do not wallog AUX file at replica */
+	if (!XLogInsertAllowed())
+		return;
+
+	snprintf(prefix, sizeof(prefix), "neon-file:%s", path);
+	if (fd < 0)
+	{
+		elog(DEBUG1, "neon: deleting contents of file %s", path);
+		/* unlink file */
+		LogLogicalMessage(prefix, NULL, 0, false, true);
+	}
+	else
+	{
+		off_t size = lseek(fd, 0, SEEK_END);
```

Comment:
worth noting in a comment that this function changes the file position. That could confuse the caller. (I don't think any of the current callers mind, but it's a hazard)

Reply:
Added the comment